### PR TITLE
flatten() function

### PR DESCRIPTION
**Pull request by Ishticode**
@djl11 
Couldn't find `flatten()`. If it is implemented with a different name then i will close this. 
If you think this is good addition then i will write up some docstring and typehints and fix the test.
Just felt it was needed at times...

### CODE REVIEW

File: ivy/functional/backends/tensorflow/manipulation.py [hunk 1]
```diff
@@ -231,3 +232,10 @@ def clip(
     else:
         ret = tf.clip_by_value(x, x_min, x_max)
     return ret
+
+
+def flatten(x):
+    shape = 1
+    for dim in x.shape:
+        shape *= dim
+    return tf.reshape(x, shape)
```
Comment (djl11) [inline ivy/functional/backends/tensorflow/manipulation.py:241]: Does TensorFlow not support `tf.reshape(x, -1)`?

Comment (djl11) [reply ivy/functional/backends/tensorflow/manipulation.py:241]: If it does not, you can do this in one line like so:

```
import math
tf.reshape(x, math.prod(x.shape))
```

**Review by djl11 — CHANGES_REQUESTED** [review]
Please respond to the comments, and then re-request a review.

File: ivy/functional/backends/tensorflow/manipulation.py (hunk 1 above)
Comment (Ishticode) [reply ivy/functional/backends/tensorflow/manipulation.py:241]: yh I will change this.

**Comment by Ishticode** [comment]
ivy.reshape(x, -1) works with all except pytorch which requires shape to be tuple.

I will just add this bit to pytorch in reshape:
```
if isinstance(shape, int)
    shape = (shape, )
```

**Comment by Ishticode** [comment]
Will reopen and merge.

**Pull request closed without merging**